### ISSUE TITLE
Some Websites Accept Dobby, others Don't?

### ISSUE THREAD

**Issue by jstacklin** [tfrommen/Dobby#8]
#### Version Information

* PHP: 7.0.21 (I think)
* WordPress: Latest
* Server OS: CentOS 7
* CHOWN access: YES
* Other plugin issues: NO

#### Steps to Reproduce
Unknown, I installed the plugin from the WP repo on two different sites on my network of sites (read: not multisite) and one works and gets rid of errors and messages, but the other Dobby ignores messages present.

#### Debugging steps taken

* Disable, then reenable plugin
* Disable, then install, then clean orphan tables
* Reinstall and reenable plugin
* Change the value of detection to 0

#### What I Expected

It to just work

#### What Happened Instead

See above in steps to reproduce

#### Plugins Installed (assume latest versions on all)

* Add From Server by Dion Hulse.
* Advanced Database Cleaner by Younes JFR..
* All In One SEO Pack by Michael Torbert.
* Contact Form 7 by Takayuki Miyoshi.
* Dobby by Thorsten Frommen.
* Duplicator Pro by Snap Creek.
* Easy WP SMTP by wpecommerce.
* EWWW Image Optimizer by Shane Bishop.
* Goodlayers Core by Goodlayers.
* Goodlayers Portfolio Post Type by Goodlayers.
* Maintenance by fruitfulcode.
* Plugins List by David Artiss.
* Tabby Responsive Tabs by cubecolour.
* Typing Effect by 93digital.
* Wordfence Security by Wordfence.
* WP Fastest Cache by Emre Vona.

**Comment by jstacklin** [comment]
Potential solution:
Edited plugin: "Duplicator Pro". Re-downloading (and uploading) will advise on if this works or not after testing. 

Potential issue: Improperly edited plugins may not "display" warnings correctly.

**Comment by tfrommen** [comment]
Hi there,

I'm not following exactly, I guess.

Are you talking about one and the same admin notice (irrelevant of what its cause is) that does get caught on one server, and does not on another?

To clarify a little what Dobby does: the plugin captures **properly rendered** admin notices, meaning whatever gets rendered on one of the admi notice hooks.

If plugins are _doing it wrong_, or also _settings errors_ and other WordPress-specific information that happens to be displayed in the _admin header_ will not get caught. This is by design.

**Comment by jstacklin** [comment]
Hey there!

So I have finally concluded that the issue was simply that the plugin (in my case, Duplicator Pro) had been modified so that there was no actual error (user could see the error, but it wasn't coded properly so WP couldn't display it as such). By re-installing the latest version of the plugin, the issue was removed. **Users with this issue for any plugin that messages don't get hidden by Dobby should try to disable, then upload the latest version of the plugin, then enable that version, finally remove the old version** (I have to do mine through FTP or SSH for some unknown, unrelated issue).

Thank you for an awesome plugin and I wish you a wonderful day!
